agent/tools/retrieval.py

```
@@ -34,6 +34,21 @@
+def _shared_embedding_id(records, mismatch_message: str):
+    """Return a stored embedding id when records share one resolved model.
+
+    Comparison uses ``validate_dataset_embedding_models`` so a tenant_model
+    UUID and a legacy ``model@instance@provider`` composite are compatible
+    when they refer to the same model. Empty or unresolved names stay
+    isolated. The returned value is an original stored ``embd_id`` (UUID or
+    composite) for ``resolve_model_config``, not a normalized base name.
+    """
+    err = validate_dataset_embedding_models(records)
+    if err:
+        raise Exception(mismatch_message)
```

Ruff (0.16.3)

[warning] Create your own exception

(TRY002)